### PR TITLE
Fix Endian issue in SipHash for s390x

### PR DESCRIPTION
**Pull request by HarryLeeIBM**
There is an Endian issue in SipHash algorithm which uses Endian dependent code, and this caused SipHash failure in Big-Endian machine like s390x. For example, the following unit test failure is caused by the issue:

`[  FAILED  ] HadoopSnappyDecoder.repeatNeedMoreInput (1 ms)`

The fix is to add unaligned Load/Store function for Little Endian which works on both Endian machines and use them in SipHash algorithm. 

### Changelog category (leave one):
- Build Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Implement a piece of code related to SipHash for s390x architecture (which is not supported by ClickHouse).

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
@HarryLeeIBM No visible bug existed in ClickHouse, we cannot classify this pull request as a bug fix.

**Comment by alexey-milovidov** [comment]
> Fixed Endian issue in SipHash algorithm.

This is an incorrect changelog entry.

**Comment by alexey-milovidov** [comment]
This will be correct:

> Implement a piece of code related to SipHash for s390x architecture (which is not supported by ClickHouse).

File: base/base/unaligned.h [hunk 1]
```diff
@@ -2,7 +2,39 @@
...
+    if constexpr (std::endian::native == std::endian::little)
+        memcpy(&res, address, sizeof(res));
+    else
+        reverseMemcpy(&res, address, sizeof(res));
+    return res;
+}
+
+template <typename T>
+inline void unalignedStoreLE(void * address,
+                           const typename std::enable_if<true, T>::type & src)
```
Comment (alexey-milovidov) [inline base/base/unaligned.h:30]: Why enable_if?

Comment (HarryLeeIBM) [reply base/base/unaligned.h:30]: Hi @alexey-milovidov , the reason is the same as described in the comment of unalignedStore(). Thanks.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**